test/features/public_asset/domain/usecase/clean_up_public_assets_interactor_test.dart

```diff
@@ -0,0 +1,447 @@
+import 'package:dartz/dartz.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:jmap_dart_client/jmap/account_id.dart';
+import 'package:jmap_dart_client/jmap/core/id.dart';
+import 'package:jmap_dart_client/jmap/core/session/session.dart';
+import 'package:jmap_dart_client/jmap/identities/identity.dart';
+import 'package:mockito/annotations.dart';
+import 'package:mockito/mockito.dart';
+import 'package:tmail_ui_user/features/manage_account/presentation/model/identity_action_type.dart';
+import 'package:tmail_ui_user/features/public_asset/domain/model/public_assets_in_identity_arguments.dart';
+import 'package:tmail_ui_user/features/public_asset/domain/state/add_identity_to_public_assets_state.dart';
+import 'package:tmail_ui_user/features/public_asset/domain/state/delete_public_assets_state.dart';
+import 'package:tmail_ui_user/features/public_asset/domain/state/remove_identity_from_public_assets_state.dart';
+import 'package:tmail_ui_user/features/public_asset/domain/usecase/add_identity_to_public_assets_interactor.dart';
+import 'package:tmail_ui_user/features/public_asset/domain/usecase/clean_up_public_assets_interactor.dart';
+import 'package:tmail_ui_user/features/public_asset/domain/usecase/delete_public_assets_interactor.dart';
+import 'package:tmail_ui_user/features/public_asset/domain/usecase/remove_identity_from_public_assets_interactor.dart';
+
+import 'clean_up_public_assets_interactor_test.mocks.dart';
+
+@GenerateNiceMocks([
+  MockSpec<RemoveIdentityFromPublicAssetsInteractor>(),
+  MockSpec<DeletePublicAssetsInteractor>(),
+  MockSpec<AddIdentityToPublicAssetsInteractor>(),
+  MockSpec<Session>(),
+  MockSpec<AccountId>(),
+])
+void main() {
+  final removeIdentityFromPublicAssetsInteractor = MockRemoveIdentityFromPublicAssetsInteractor();
+  final deletePublicAssetsInteractor = MockDeletePublicAssetsInteractor();
+  final addIdentityToPublicAssetsInteractor = MockAddIdentityToPublicAssetsInteractor();
+  final cleanUpPublicAssetsInteractor = CleanUpPublicAssetsInteractor(
+    removeIdentityFromPublicAssetsInteractor,
+    deletePublicAssetsInteractor,
+    addIdentityToPublicAssetsInteractor,
+  );
+
+  group('clean up public assets interactor test:', () {
+    final session = MockSession();
+    final accountId = MockAccountId();
+
+    test(
+      'should destroy unused public assets '
+      'when there are picked public assets that are deleted',
+    () async {
+      // arrange
+      final identityId = IdentityId(Id('id-1'));
+      const identityActionType = IdentityActionType.edit;
+      final newlyPickedPublicAssetIds = [Id('public-asset-1'), Id('public-asset-2')];
```

Comment:
please use variable name for these to be more clear in context

Reply:
done